src/server/controllers/match.controller.js

```diff
@@ -9,76 +11,196 @@ import mongoose from 'mongoose';
+      if (err) {
+        matchCallback.fail.errPoint = 'RequestMentoring - Error occurred when finding montor_id';
+        matchCallback.fail.err = err;
+        res.json(matchCallback.fail);
+      }
+      else {
+        if (mentorDoc.length === 0) {  //if mentor_id is not existing
+          matchCallback.fail.errPoint = 'RequestMentoring - Cannot found mentor.';
+          matchCallback.fail.err = err;
+          res.json(matchCallback.fail);
+        }
+        else {
+          Match.find({mentor_id: matchData.mentor_id, mentee_id: matchData.mentee_id}, (err, matchDoc) => { //find match is existing
+            if (err) {
+              matchCallback.fail.errPoint = 'RequestMentoring - Error occurred when finding matchData';
+              matchCallback.fail.err = err;
+              res.json(matchCallback.fail);
+            }
+            else {
+              if (matchDoc.length !== 0) {  //if match is already existing
+                matchCallback.fail.errPoint = 'RequestMentoring - Match is Already exists';
+                matchCallback.fail.err = err;
+                res.json(matchCallback.fail);
+              }
+              else {
+                sendRequestEmail(res, mentorDoc[0].email, req.session.email, matchData.content, (result) => {  //try to send email
+                  if (result === false) {
+                    matchCallback.fail.errPoint = 'RequestMentoring - Fail to send email.';
+                    matchCallback.fail.err = err;
+                    res.json(matchCallback.fail);
+                  }
+                  else {
+                    match.save((err) => {
+                      if (err) {
+                        matchCallback.fail.errPoint = 'RequestMentoring - Saving MatchData';
+                        matchCallback.fail.err = err;
+                        res.json(matchCallback.fail);
+                      }
+                      else {
+                        res.json(matchCallback.successSendMail);
+                      }
+                    });
+                  }
+                });
 
-  matchData.save((err) => {
-    if (err) {
-      sampleFailResult.errPoint = 'RequestMentoring - Saving MatchData';
-      sampleFailResult.err = err;
-      res.json(sampleFailResult);
-    } else {
-      // Todo:
-      // 1. update mentor and mentee's request db
+              }
+            }
+          });
+        }
+      }
+    });
 
-      // 2. Send Request Mail to Mentor
-      sendRequestEmail(res, req.body.mentorId, req.body.menteeId, req.body.content);
-    }
-  });
+  }
+  else {
+    res.status(400).json(authCallback.failAuth);
+  }
 }
 
-// The mentor accepted the mentoring request
-export function acceptRequest(req, res, next) {
-  // Todo:
-  // update mentor and mentee's request db
+export function getMyActivity(req, res, next) {
+  if (req.session._id) {
+    let activityData = {};
+    //status of match {pending:2, accepted:1, rejected:0}
```

Comment:
It would be better use constant rather than 'magic number'.

e.g) const pending = 2